src/main/java/vazkii/patchouli/api/IRenderingStyle.java

```diff
@@ -0,0 +1,25 @@
+package vazkii.patchouli.api;
+
+import net.minecraft.util.text.Style;
+
+import java.util.function.UnaryOperator;
+
+/**
+ * Represents a style (or style stack) when rendering book text.
+ */
+public interface IRenderingStyle {
```

Comment:
why not just `StyleStack`

Reply:
there's nominally some additional data attached in implementation (e.g. span stuff), so I left the name more general

Reply:
stylestack works though